onnxruntime/core/providers/cpu/generator/constant_of_shape.h

```diff
@@ -9,13 +9,10 @@
 
 namespace onnxruntime {
 
-class ConstantOfShape final : public OpKernel {
- public:
-  explicit ConstantOfShape(const OpKernelInfo& info);
-
-  Status Compute(OpKernelContext* ctx) const override;
+class ConstantOfShapeBase {
+ protected:
+  ConstantOfShapeBase(const OpKernelInfo& info);
 
- private:
   ONNX_NAMESPACE::TensorProto_DataType tensor_type_;
```

Comment:
Members should remain private and there should be protected accessors for them as needed.